src/test/java/com/conveyal/gtfs/GTFSTest.java

```diff
@@ -456,10 +481,43 @@ private void assertThatImportedGtfsMeetsExpectations(
                     LOG.info("Database record satisfies expectations.");
                     foundRecord = true;
                     break;
+                } else {
+                    LOG.error("Persistence mismatch on record {}", numRecordsSearched);
                 }
             }
             assertThatPersistenceExpectationRecordWasFound(numRecordsSearched, foundRecord, fieldsWithMismatches);
         }
+        // Check that error expectations match errors stored in database.
+        if (errorExpectations != null && errorExpectations.length > 0) {
+            LOG.info("Checking {} error expectations", errorExpectations.length);
+            // select all entries from error table
+            String sql = String.format("select * from %s.errors", namespace);
+            LOG.info(sql);
+            Iterator<ErrorExpectation> errorExpectationIterator = Arrays.stream(errorExpectations).iterator();
+            ResultSet rs = connection.prepareStatement(sql).executeQuery();
+            int errorCount = 0;
+            while (rs.next()) {
+                errorCount++;
+                String errorType = rs.getString("error_type");
+                String entityType = rs.getString("entity_type");
+                String entityId = rs.getString("entity_id");
+                String badValue = rs.getString("bad_value");
+                // Skip error expectation if not exists. But continue iteration to count all errors.
+                if (!errorExpectationIterator.hasNext()) continue;
+                ErrorExpectation errorExpectation = errorExpectationIterator.next();
+                LOG.info("Checking error {} (found={} expected={})", errorCount, errorType, errorExpectation.errorType);
+                assertThat(errorExpectation.errorType.toString(), equalTo(errorType));
+                if (errorExpectation.entityType != null) assertThat(errorExpectation.entityType, equalTo(entityType));
+                if (errorExpectation.entityId != null) assertThat(errorExpectation.entityId, equalTo(entityId));
+                // FIXME: How should we handle the expectation that the bad value expected matches null?
+                if (errorExpectation.badValue != null) assertThat(errorExpectation.badValue, equalTo(badValue));
+            }
+            if (errorExpectations.length < errorCount) {
+                LOG.warn("More errors stored in database ({}) than expectations ({}) found in test.", errorCount, errorExpectations.length);
```

Comment:
Can you add an assertion for this condition? This should result in an error to make sure there are no false positives.